homeassistant/components/zwave/api.py

```diff
@@ -0,0 +1,99 @@
+"""API class to give info to the Z-Wave panel."""
+
+import logging
+import homeassistant.core as ha
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import HTTP_NOT_FOUND
+from . import const
+
+_LOGGER = logging.getLogger(__name__)
+
+ZWAVE_NETWORK = 'zwave_network'
+
+
+class ZWaveNodeGroupView(HomeAssistantView):
+    """View to return the nodes group configuration."""
+
+    url = r"/api/zwave/groups/{node_id:\d+}"
+    name = "api:zwave:groups"
+
+    @ha.callback
+    def get(self, request, node_id):
+        """Retrieve groups of node."""
+        # pylint: disable=import-error
+        from openzwave.group import ZWaveGroup
+
+        nodeid = int(node_id)
+        hass = request.app['hass']
+        network = hass.data.get(ZWAVE_NETWORK)
+        node = network.nodes.get(nodeid)
+        if node is None:
+            return self.json_message('Node not found', HTTP_NOT_FOUND)
+        groupdata = node.groups_to_dict()
```

Comment:
Why fetch the groups as dict, then iterate over the keys and construct group objects again?

Seems like `node.groups` will give  you a list of groups too? https://github.com/OpenZWave/python-openzwave/blob/master/src-api/openzwave/node.py#L289